src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -216,11 +230,23 @@ private Predicate addDateFilter(FiltersActivityDTO filtersActivityDTO, Predicate
         Join<ActivityInfo, OrganizationMatch> matchJoin = root.join(ORGANIZATION_MATCHES, JoinType.LEFT);
         predicate = cb.and(predicate, cb.equal(matchJoin.get(HIDDEN), filtersActivityDTO.getHiddenFilter()));
 
-        if (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList())) {
+        boolean hasPartnerFilters = (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList()));
+        if (partnerRecords || hasPartnerFilters) {
             Join<OrganizationMatch, Organization> matchOrgJoin = matchJoin.join(PARTNER_VERSION, JoinType.LEFT);
-            Join<Organization, SystemAccount> accountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
-
-            predicate = cb.and(predicate, accountJoin.get(ID).in(filtersActivityDTO.getPartnerFilterList()));
+            Join<Organization, SystemAccount> matchAccountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
+            if (partnerRecords) {
+                predicate = cb.and(predicate, cb.or(cb.equal(root.get(ACCOUNT_ID), ownerId),
+                    cb.notEqual(matchAccountJoin.get(ID), ownerId)));
+                if (hasPartnerFilters) {
+                    Join<Organization, SystemAccount> accountJoin = orgJoin.join(ACCOUNT);
+                    predicate = cb.and(predicate, cb.or(cb.equal(root.get(ACCOUNT_ID), ownerId),
```

Comment:
this filter will not work, it will display all organizations of current user, it will not filter them by matched account
remove the `cb.or(cb.equal(root.get(ACCOUNT_ID), ownerId)`